### ISSUE TITLE
GPU not used for feature_extraction when options are specified in .ini project file

### ISSUE THREAD

**Issue by ClaudioCimarelli** [colmap/colmap#1110]
**Describe the bug**
When I use the `feature_extractor ` as `colmap feature_extractor  --database_path $DATASET_PATH/database.db --image_path $DATASET_PATH/images` with default params, the GPU gets used as I notice the processes in nvidia-smi. However, when I run the same command specifying the `project_path` to a local project  .ini file the GPU is not activated. 
The project file is generated with `colmap project_generator` and I modified it by simply commenting all the lines not referring to valid options for the feature extraction step. Also, I tried converting the "true"/"false" values into "1"/"0" resp.. Of course, nothing worked.

**Platform**
As an additional detail, I am running the software on a HPC. Therefore, I am forced to use the command-line interface and cannot try the GUI.

Thanks.

**Comment by tsattler** [comment]
The GPU implementation used by Colmap does not support estimating affine shapes and domain size pooling. As such, feature extraction is done on the CPU if one of the two options is activated (see also #1105 ).

**Comment by ClaudioCimarelli** [comment]
Thanks a lot for your answer. May I ask if it is worth using those additional options while, on the other hand, is advised to use GPU feature extraction for improved robustness? 
From https://colmap.github.io/tutorial.html:
>  GPU version is favorable as it has a customized feature detection mode that often produces higher quality features in the case of high contrast images.

**Comment by potblender** [comment]
I've downloaded COLMAP.
When running 'Run Tests.bat' there shows no errors. (Green OK etc)
Opening the program using the 'Colmap.bat' and loading the images, the Feature Extracting process just slows the laptop down. Any ideas?
Looked in to the Graphics card settings (NVIDIA GeForce 540M), but can only assume thats ok, as it works on MeshRoom and Blender.
Also, ZEPHYR 3D FREE produces good models but the file cannot be exported for use in other editing programs?
Any help much appreiciated!

**Comment by tsattler** [comment]
@ClaudioCimarelli 
> Thanks a lot for your answer. May I ask if it is worth using those additional options while, on the other hand, is advised to use GPU feature extraction for improved robustness?

Using these parameters can help in certain cases as it should lead to features that are a bit more stable to viewpoint changes. But it can also hurt since they might lead to fewer features. I'd say try them if the GPU does not give you satisfactory results.


@potblender 
> Opening the program using the 'Colmap.bat' and loading the images, the Feature Extracting process just slows the laptop down. Any ideas?

Is it still extracting features (there should be some output to Colmap's console)? Feature extraction can take time and resources, depending on the number of images and the image resolution.

**Comment by potblender** [comment]
When running the Feature Extraction the laptop seems to just slow down....even just moving the mouse around!
I'm sure Colmap has work before (i've lost the data!) and the Feature Extraction was quite quick (a few seconds!).
It might of been smaller file images (off mobile phone) and now been trying better quality photos off a dslr and/or Canon S95 so it might be that or low memory?
Really looking for best/easiest route (with my system) to produce a 3D mesh to get to a end point in be able to print a 3D model.
I'm new to photogrammetry and did have issues with DepthMap node in Meshroon which I seem to have gotten around!
It's frustrating when I can see a top quality (and coloured) looking meshes (done quite quickly!) in Zephyr 3DFree (but cannot use their files in other programs!) and compared to
Colmap (very slow) and Meshroom (slow) using same images!
Hopefully, I will figure it out.....with some guidence lol
Thanks for your input!

**Comment by tsattler** [comment]
If the laptop slows down, then likely you are running out of memory (either RAM or GPU memory). That would be one thing to look into.

**Comment by potblender** [comment]
Yeah, I was being warned of low memory when first starting, so cleared some space and upped the RAM from 6GB to 8GM (my max!).
Been okay lately!
I'm sure it will be something quite simple I'm missing, for the number/quality of images v system I'm using.
Will have another go.....
Thanks again!

**Comment by potblender** [comment]
Also....
While running Meshroom the laptop doesn't slow down as it does in Colmap?!
Would of thought there wouldn't be 'that' much difference in system usage??

**Comment by tsattler** [comment]
I have never used Meshroom and cannot comment on it. I am using Colmap for research purposes as has nice clean code and can be easily modified.

**Comment by Claudio9701** [comment]
In the [Feature Detection and Extraction Section of the COLMAP Tutorial](https://colmap.github.io/tutorial.html#feature-detection-and-extraction In the third paragraph) they mention the following: 

> "... The **GPU version requires an attached display**, while the **CPU version is recommended for use on a server**. ..."
